### PR TITLE
Updated programming tutorials

### PR DESCRIPTION
**Pull request by AbhignyaTayi**
## What does this PR do?
Added programming tutorial for data science and GLSL

### Why is this valuable (or not)?
it is valuable because the tutorials are staring from basics by experienced tutors and easy for anyone to understand

### How do we know it's really free?
It is free of cost

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: more/free-programming-interactive-tutorials-en.md [hunk 1]
```diff
@@ -136,7 +136,7 @@
 * [Essential Linear Algebra for Data Science and Machine Learning](https://www.kdnuggets.com/2021/05/essential-linear-algebra-data-science-machine-learning.html) - KDnuggets
 * [Interactive Linear Algebra](https://textbooks.math.gatech.edu/ila/) - Dan Margalit, Joseph Rabinoff (HTML, PDF)
 * [Top 10 Data Science Projects for Beginners - KDnuggets](https://www.kdnuggets.com/2021/06/top-10-data-science-projects-beginners.html)
-
+*[Data science basics](https://www.w3schools.com/datascience/)
```
Comment (mohdahsanrazakhan) [inline more/free-programming-interactive-tutorials-en.md:139]: Move line number `139` to line `132` for alphabetical ordering.

File: more/free-programming-interactive-tutorials-en.md [hunk 2]
```diff
@@ -159,6 +159,7 @@
 ### GLSL
 
 * [The Book of Shaders](https://thebookofshaders.com)
+* [GLSL Basics By Jacobo Rodriguez Villar](https://www.opengl.org/sdk/docs/tutorials/TyphoonLabs/Chapter_2.pdf)
```
Comment (mohdahsanrazakhan) [inline more/free-programming-interactive-tutorials-en.md:162]: Swap line number `162` and `161`

File: more/free-programming-interactive-tutorials-en.md (hunk 1 above)
Comment (mohdahsanrazakhan) [inline more/free-programming-interactive-tutorials-en.md:139]:
```suggestion
* [Data science basics](https://www.w3schools.com/datascience/)
```

**Review by mohdahsanrazakhan — COMMENTED** [review]
Fix your code.

File: more/free-programming-interactive-tutorials-en.md [hunk 3]
```diff
@@ -129,6 +129,7 @@
 
 ### Data Science
 
+* [Data science basics](https://www.w3schools.com/datascience/)
```
Comment (eshellman) [inline more/free-programming-interactive-tutorials-en.md:132]:
```suggestion
* [Data Science Tutorial](https://www.w3schools.com/datascience/)
```
title from resource

**Comment by eshellman** [comment]
duplicate

**Pull request closed without merging**